### ISSUE TITLE
Any plans for F-Droid repo?

### ISSUE THREAD

**Issue by r7l** [anyproto/anytype-kotlin#386]
### Have you read a contributing guide?

- [X] I have read CONTRIBUTING.md
- [X] I have searched the existing requests and didn't find any that were similar
- [X] I have considered creating a pull request instead and want to proceed

### Clear and concise description of the problem

Some of your targeted users (me for example) might run non Google based phones. This can easily be archived by installing a custom rom without installing additional Google support. Those phones will lack a Play Store. The app currently seems only be available from Google Play Store.

### Suggested solution

Submit the app to [F-Droid store](https://f-droid.org).

**Comment by visika** [comment]
I agree, anyway in the meantime I'm installing Anytype and tracking updates with [Obtainium](https://github.com/ImranR98/Obtainium)

**Comment by Atrate** [comment]
Maybe @IzzySoft could help with a stop-gap measure (the IzzyOnDroid repo)

**Comment by shuvashish76** [comment]
https://gitlab.com/IzzyOnDroid/repo/-/wikis/Inclusion%20Policy

<details><summary>Details</summary>
<p>

> Running on private resources (no funding), IzzyOnDroid usually reserves up
to 30 megabytes per app (exceptions are being made for some larger apps, so this is
considered as rule-of-thumb). That's at the same time the upper size limit for
single .apk files. If multiple files can fit in this limit, the repo holds up
to 3 versions.

</p>
</details>

**Comment by shuvashish76** [comment]
<details><summary>Details</summary>
<p>

> To determine which licenses are FLOSS, We defer to widely trusted organizations that have a proven track record. Specifically, we acknowledge these standards: [DFSG](https://wiki.debian.org/DFSGLicenses), [FSF](https://www.fsf.org/licensing), [GNU](https://www.gnu.org/licenses/license-list.html), and [OSI](https://opensource.org/licenses) (read a quick overview of them all on [SPDX](https://spdx.org/licenses/)).

</p>
</details>

Source: https://f-droid.org/en/docs/Inclusion_Policy/

I searched the mentioned sources for **[Any Source Available License 1.0](https://github.com/anyproto/anytype-kotlin/blob/main/LICENSE.md)** but no results. So CMIIW @linsui technically it's not FOSS (at least to F-Droid standard) hence can't be included to F-Droid ?

**Comment by linsui** [comment]
> but only (a) for Non-Commercial Use, or (b) for Commercial Use in Allowed Networks.

This is obviously not FOSS.

**Comment by IzzySoft** [comment]
> I searched the mentioned sources for [Any Source Available License 1.0](https://github.com/anyproto/anytype-kotlin/blob/main/LICENSE.md?rgh-link-date=2024-07-08T02%3A20%3A11Z) but no results.

Not even *mentioned* at [SPDX](https://spdx.org/licenses/) – so no, this cannot be considered libre I'm afraid.

**Comment by linsui** [comment]
> licences protecting free software by not allowing commerce

This is not free software. Freedom 0 of the 4 freedoms defined by FSF is the freedom to run the program as you wish, for any purpose.

**Comment by twiguard** [comment]
yeah, it's not absolutely free, and it's not fancy Free™, it is free though for any living individual to use, make changes, learn from and distribute
as for for-profit companies, I have no need to give them any more compassion than they have for any of us, which is factually zero
I don't see, why we, the living people, should argue about defending their rights and possibilities which are already far outstretched

now more on topic:
any idea what it would take to host a basic f-droid repo?

**Comment by IzzySoft** [comment]
If you want to use F-Droid.org's framework, see e.g. [Setting up a simple binary F-Droid repo](https://android.izzysoft.de/articles/named/fdroid-simple-binary-repo) and the follow-up article [Extend the „simple binary repo“: Screenshots & more](https://android.izzysoft.de/articles/named/fdroid-simple-binary-repo-2). An easier approach would probably be [apkrepotool](https://codeberg.org/obfusk/apkrepotool) – but that is currently unmaintained.

**Comment by taxmeifyoucan** [comment]
+1 just learned that it's source available, not FOSS. Maintainer should consider fixing their license first and switch to GPL

**Comment by trymeouteh** [comment]
I am fine with this app being available on the official fdroid repo, and not needing a self hosted fdroid repo for this app

**Comment by IzzySoft** [comment]
Which cannot happen without a proper FOSS license. The current one is not even listed [at spdx](https://spdx.org/licenses/), not to speak of being approved by OSI or FSF.

Same goes for the [IzzyOnDroid repo](https://apt.izzysoft.de/fdroid) – though the APK size rules out that option already. Which only leaves a self-hosted (or other non-FOSS) repo then.

**Comment by KaKi87** [comment]
> The current one is not even listed at spdx

Of course not, because it's their own license : the "Any" in "Any Source Available License 1.0" means "The Any Association".

They also provide some additional explanations at https://legal.any.coop basically saying they use this license to forbid commercial use.

I don't know why they didn't choose CC-BY-NC-SA though.

**Comment by IzzySoft** [comment]
CC licenses are not intended for code, but rather for creative works like texts or graphics. Apart from that would the NC part still be a show-stopper for F-Droid, as it limits freedom-0 (the "freedom to run the program as you wish, for any purpose", see [The Four Essential Freedoms of Free Software](https://en.wikipedia.org/wiki/The_Free_Software_Definition#The_Four_Essential_Freedoms_of_Free_Software)). A bit controversial, I admit; not wanting the own free work monetized (or otherwise abused) by others is totally understandable IMHO.

**Comment by KaKi87** [comment]
Well, SPDX is for software and CC licenses are part of SPDX.

I also see [NPOSL](https://spdx.org/licenses/NPOSL-3.0.html) ?

**Comment by trymeouteh** [comment]
Will it possible to dual license the android app with a friendly license which can be accepted by F-Droid?

**Comment by alensiljak** [comment]
Discussion in the forum:
https://community.anytype.io/t/add-anytype-to-the-f-droid-store/9613/
